### PR TITLE
Add lua test to the repository.

### PR DESCRIPTION
**Pull request by alex4o**
A very simple set of test for checking lua.

### CODE REVIEW

**Comment by alex4o** [comment]
This should fail

bors try

**Comment by alex4o** [comment]
bors try 

This again should fail

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

**Comment by alex4o** [comment]
bors try

File: .buildbot.sh [hunk 1]
```diff
@@ -4,22 +4,49 @@ set -e
...
-	sh rustup.sh --default-host x86_64-unknown-linux-gnu \
-		--default-toolchain stable \
-		--no-modify-path \
-		--profile minimal \
-		-y
-	source $HOME/.cargo/env
-fi
-cargo test
+pushd tests
+	if ! [ -x "$(command -v cargo)" ]; then
```
Comment (ltratt) [inline .buildbot.sh:20]: I don't think these lines should be indented?

File: .buildbot.sh [hunk 2]
```diff
@@ -4,22 +4,49 @@ set -e
...
+		source $HOME/.cargo/env
+	fi
+	cargo test
+popd 
+
+tmpdir=/tmp/lua-$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 10 | head -n 1)/
+
+mkdir -p $tmpdir
+pushd $tmpdir
+	git clone --depth 1 -b v5.4.0 https://github.com/lua/lua.git .
```
Comment (ltratt) [inline .buildbot.sh:36]: Or these.

**Comment by ltratt** [comment]
Rather than `pushd`/`pop` we've generally preferred `cd X` `cd ..` because `pushd` isn't portable.

**Comment by ltratt** [comment]
I'm also a bit allergic to directories with `.` in their names -- can we call that directory something like `lua_tests`?

File: .buildbot.sh (hunk 1 above)
Comment (alex4o) [reply .buildbot.sh:20]: It allows the ability to visually determine which things run in a different directory then the original one.

**Comment by alex4o** [comment]
Done. Is this looking better?

**Comment by ltratt** [comment]
Please squash.

**Comment by alex4o** [comment]
a bit late but done

**Pull request merged**